dvc/remote/ssh/__init__.py

```diff
@@ -207,8 +250,6 @@ def _upload(self, from_file, to_info, name=None, no_progress_bar=False):
 
     @contextmanager
     def open(self, path_info, mode="r", encoding=None):
-        assert mode in {"r", "rt", "rb"}
```

Comment:
You removed assert, but you didn't implement new modes. BTW, it would be easier for you to implement `'wb'` mode then `'w'`, since paramiko files are always binary.